src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java

```diff
@@ -103,10 +103,9 @@ public BitbucketGitSCMBuilder(@NonNull BitbucketSCMSource scmSource, @NonNull SC
         this.scmSource = scmSource;
 
         String serverURL = scmSource.getServerUrl();
-        AbstractBitbucketEndpoint endpoint = BitbucketEndpointConfiguration.get().findEndpoint(serverURL);
-        if (endpoint == null) {
-            endpoint = new BitbucketServerEndpoint(null, serverURL, false, null);
-        }
+        AbstractBitbucketEndpoint endpoint = BitbucketEndpointConfiguration.get()
+                .findEndpoint(serverURL)
```

Comment:
serverURL is never null because scmSource.getServerUrl() return a non null value (default is cloud).
normalizeServerUrl is useless here